src/error-snippets/source-maps.ts

```diff
@@ -0,0 +1,45 @@
+import { SourceMapConsumer, type BasicSourceMapConsumer } from "source-map";
+import url from "url";
+import { SOURCE_MAP_URL_COMMENT } from "./constants";
+import { softFileURLToPath, getSourceCodeFile } from "./utils";
+import type { SufficientStackFrame, ResolvedFrame } from "./types";
+
+export const extractSourceMaps = async (
+    fileContents: string,
+    fileName: string,
+): Promise<BasicSourceMapConsumer | null> => {
+    const sourceMapsStartIndex = fileContents.indexOf(SOURCE_MAP_URL_COMMENT);
+    const sourceMapsEndIndex = fileContents.indexOf("\n", sourceMapsStartIndex);
+
+    if (sourceMapsStartIndex === -1) {
+        return null;
+    }
+
+    const sourceMapUrl =
+        sourceMapsEndIndex === -1
+            ? fileContents.slice(sourceMapsStartIndex + SOURCE_MAP_URL_COMMENT.length)
+            : fileContents.slice(sourceMapsStartIndex + SOURCE_MAP_URL_COMMENT.length, sourceMapsEndIndex);
+
+    const sourceMaps = await getSourceCodeFile(url.resolve(fileName, sourceMapUrl));
```

Comment:
`sourceMapUrl` could either be relative, absolute or `data:application/json;base64,...`

> https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/edit#heading=h.lmz475t4mvbx

After `url.resolve` we could get:
- absolute path (which we could fs.readFile)
- embedded (which would be handled by fetch)